apps/designer/app/canvas/hovered-instance-connector.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect } from "react";
```

Comment:
lots of new logic written in this file without taking it out of react and only using minimal amount of react specific code as we discussed we need to do on canvas

I wonder if we actually agreed on the approach, because if now I would like to hear a strategy how we can get to a cross-framework canvas otherwise :)

Reply:
Alright, though it's same logic as before + scroll fix

Reply:
since you already know what to test there, its simple right now for you to migrate